agent/exploits/cve_2024_47374.py

```diff
@@ -0,0 +1,46 @@
+"""Agent Asteroid implementation for CVE-2024-47374"""
+
+import re
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+# Vulnerable versions: all versions up to and including 6.5.0.2
+VULNERABLE_VERSIONS = [
+    r"(?P<major_0_to_5>[0-5])\.(?P<minor>[0-9]+)(\.(?P<patch>[0-9]+))?",  # Covers all versions from 0.0.0 to 5.x.x
+    r"(?P<major_6_0_to_4>6)\.(?P<minor_0_to_4>[0-4])(\.(?P<patch>[0-9]+))?",  # Covers all 6.0.x to 6.4.x
+    r"(?P<major_6_5>6)\.5\.0(\.(?P<patch_0_to_2>[0-2]))?",  # Covers 6.5.0, 6.5.0.1, and 6.5.0.2
```

Comment:
If the version is 6.5.0.6 , the regex will match the 6.5.0 even if it's not vulnerable , and also what does the P major thing in the regex ?

Reply:
You're right, I updated the regex. As for the `P major thing` it's called named capturing groups.

![image](https://github.com/user-attachments/assets/81c66b91-111c-401a-bebf-2696d0c6e417)

In our case: 

- `(?P<major_0_to_5>[0-5])`: Captures the major version (0-5).
- `(?P<minor>[0-9]+)`: Captures the minor version.
- `(?P<patch>[0-9]+)`: Captures the patch version.